info_test.go

```diff
@@ -160,8 +160,11 @@ func TestProgramInfo(t *testing.T) {
 func TestProgramInfoProc(t *testing.T) {
 	spec := fixupProgramSpec(basicProgramSpec)
 	prog := mustNewProgram(t, spec, nil)
+	info, err := prog.Info()
```

Comment:
The point of this test is to test procinfo in isolation, since it's supported on older kernels, and objinfo isn't. Adding prog.Info() and the subsequent skip will reduce coverage on those kernels.

Just instantiate a new ProgramInfo here and populate it using `readProgramInfoFromProc`.

Also, validateProgInfo needs to be extended to check for memlock, otherwise we're not testing the new field anywhere.

Reply:
Done, use a new ProgramInfo and add check for memelock field